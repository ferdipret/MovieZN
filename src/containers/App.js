import React, { Component } from 'react'
import Input from 'components/Input'

class App extends Component {
  constructor(props) {
    super(props)

    this.state = {
      // This will hold our search value
      searchInputValue: '',
      /**
       * Our centent container will have 3 states:
       * 'discover': This will be the default, with some filter options
       * 'search':   When the user makes a search, we will show relavant results
       * 'select':   If an item has been selected, we want to display more info for that item
       */
      containerState: 'discover',
    }
  }

  render() {
    const { searchInputValue } = this.state

    return (
      <div>
        <div>MovieZN</div>
        <Input
          onSearchInputChange={this.handleSearchInputChange}
          value={searchInputValue}
        />
      </div>
    )
  }

  handleSearchInputChange = e =>
    this.setState({ searchInputValue: e.target.value })
}

export default App
