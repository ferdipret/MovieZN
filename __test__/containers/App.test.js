import React from 'react'
import renderer from 'react-test-renderer'
import { mount, find, simulate } from 'enzyme'

import App from 'containers/App'

// We should make sure the app doesn't have any problems
// on the initial render
describe('App', () => {
  test('should render correctly', () => {
    const rendered = renderer.create(<App />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})

describe('App/Input', () => {
  test('should update state when onChange is called', () => {
    const wrapper = mount(<App />)
    const input = wrapper.find('.search-input')

    input.simulate('change', { target: { value: 'New Value' } })

    expect(wrapper.state().searchInputValue).toBe('New Value')
  })
})
