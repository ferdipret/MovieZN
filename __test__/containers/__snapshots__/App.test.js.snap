// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="sc-htpNat bItXhR"
  style={
    Object {
      "backgroundColor": "hsl(0, 0%, 13%)",
    }
  }
>
  <div
    className="sc-htpNat jIYwWs"
    style={undefined}
  >
    MovieZN
  </div>
  <div
    className="sc-htpNat iYrbMG"
    style={undefined}
  >
    <form
      className="sc-bwzfXH jFkwyO"
      onSubmit={[Function]}
    >
      <input
        className="search-input"
        onChange={[Function]}
        value=""
      />
      <button
        className="form-submit-btn hidden"
        type="submit"
      />
    </form>
  </div>
</div>
`;
